src/core/server/elasticsearch/client/configure_client.ts

```diff
@@ -128,7 +128,9 @@ export function getRequestDebugMeta(event: DiagnosticResult): {
   };
 }
 
-const addLogging = (client: Client, logger: Logger) => {
+const addLogging = ({ client, type, logger }: { client: Client; type: string; logger: Logger }) => {
+  const queryLogger = logger.get('query', type);
+  const deprecationLogger = logger.get('deprecation', type);
```

Comment:
With the increased scope of this function, it can now probably be extracted to its own file and own UTs.